books/build/make_cert_help.pl

```diff
@@ -831,15 +831,16 @@ sub extract_hostname_from_outfile
 	($STEP eq "pcertifyplus") ? "PROVISIONAL CERTIFICATION+" :
 	($STEP eq "convert")  ? "PCERT0->PCERT1 CONVERSION" :
 	($STEP eq "complete") ? "PCERT1->CERT COMLETION" : "UNKNOWN";
-    print "**$taskname FAILED** for $dir$file.lisp\n\n";
-    system("tail -300 $outfile | sed 's/^/   | /'") if $outfile;
-    print "\n\n";
+    print "\n**$taskname FAILED** for $dir$file.lisp\n\n" .
+        ($outfile ? `tail -300 $outfile | sed 's/^/   | /'` : "") .
+        "\n**$taskname FAILED** for $dir$file.lisp\n\n";
 
     if ($ON_FAILURE_CMD) {
-	system($ON_FAILURE_CMD);
+        print "\n-- Executing ON_FAILURE_CMD='$ON_FAILURE_CMD' for $dir$file.lisp\n\n" .
+            `$ON_FAILURE_CMD` .
```

Comment:
What do people put in `$ON_FAILURE_CMD`? If it prints stuff to `stderr`, then this won't be atomic anymore (and in fact the `stderr` output may show up before it prints "Executing ON_FAILURE_CMD=..."). It seems like writing `` `{ $ON_FAILURE_CMD ; } 2>&1` `` ought to work, but maybe I'm missing some weird corner case.

Reply:
Personally I forgot that ON_FAILURE_CMD existed. I don't know that anyone uses it.  Looks like Jared added it in 2011.